src/Evolution/DiscontinuousGalerkin/NormalVectorTags.hpp

```diff
@@ -0,0 +1,38 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <optional>
+#include <unordered_map>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// Tags used on the interior faces of the elements
+namespace evolution::dg::Tags::InternalFace {
+/// The magnitude of the unnormalized normal covector to the interface
+struct MagnitudeOfNormal : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+
+/// The normal covector to the interface
+template <size_t Dim>
+struct NormalCovector : db::SimpleTag {
+  using type = tnsr::i<DataVector, Dim, Frame::Inertial>;
+};
+
+/// The normal covector and its magnitude for all faces of an element.
+///
+/// The combined tag is used to make the allocations be in a Variables
+template <size_t Dim>
+struct NormalCovectorAndMagnitude : db::SimpleTag {
+  using type = std::unordered_map<
+      Direction<Dim>, std::optional<Variables<
+                          tmpl::list<MagnitudeOfNormal, NormalCovector<Dim>>>>>;
```

Comment:
I don't see that being worth the effort. This is all a detail of the evolution code. As a user you basically should never use any of these.

Reply:
"Detail of the evolution code" sounds pretty broad :D I suppose it depends on the definition of "user" as @moxcodes pointed out in a recent PR. If you consider it detail for a particular action it should be in the corresponding file, or at least in a `detail` namespace. As it stands now it's part of the public interface 🤷‍♂️

Are you making your own tags for the unit normal and its magnitude only to have shorter names than the existing `Normalized<UnnormalizedFaceNormal>` and `Magnitude<UnnormalizedFaceNormal>`? I remember bringing this up a while back and the answer was "just use the tags, redefining shorter ones isn't worth it". A reasonable alternative that I can see would be to just add type aliases in `domain::Tags`.

Reply:
It's only used in `ComputeTimeDerivative` and the lifting action. That's not really broad. I need to share it between those files, and I don't want to have one 10k line file.

I intentionally don't want to use those takes because I don't want to debug things where the wrong normal vectors (obtained via the interface compute tags are used). Basically, I either make incremental changes like this and we have to have some duplication regardless, or I submit one 25k line PR. I'm trying to replace the current ways of compute things one step at a time because that means my PRs are "only" 500-1000 lines.